### PR TITLE
Removing custom ProblemDetails Converters

### LINKED ISSUE #44132: Remove (Http)(Validation)ProblemDetails converters
Hi, I think this may be related, but the casing for ProblemDetails is also inconsistent. ProblemDetails itself defaults to camel case, but as you can see above when describing the "Summary" field, it's capitalized. Shouldn't that be lower case, assuming @TanvirArjel is using the default naming policy?

I've also noticed ProblemDetails does not obey `JsonOptions` settings, e.g., setting

```
services.Configure<JsonOptions>(opt =>
{
    opt.JsonSerializerOptions.PropertyNamingPolicy = null; // don't convert to camel case
});
```

should give a response like

```
{
    "Type": "https://tools.ietf.org/html/rfc7231#section-6.5.4",
    "Title": "Not Found",
    "Status": 404,
    "Detail": "Account 1 not found.",
    "TraceId": "00-50f63c5d7921484e4797d82687f43033-915b95474bf55ff2-00"
}
``` 

but it does not, at least from what I can tell. Maybe this is by design since the spec (https://www.rfc-editor.org/rfc/rfc7807) explicitly lists the properties lower/regular camel case, but it also doesn't seem to explicitly call out camelcase must be used. I think anyone who is using problem details should be able to use whatever casing their API uses already rather than being forced to either a) have inconsistent casing returned or b) use camel case.

_Originally posted by @jchoca in https://github.com/dotnet/aspnetcore/issues/43261#issuecomment-1248777800_

**Repro:**: https://github.com/jchoca/ProblemDetailsJsonSettings

### PR DESCRIPTION
**Pull request by brunolins16**
Fix https://github.com/dotnet/aspnetcore/issues/44132

### CODE REVIEW

**Comment by brunolins16** [comment]
@dotnet/minimal-apis could you please review? 🙏

File: src/Mvc/Mvc.Core/test/Infrastructure/DefaultApiProblemDetailsWriterTest.cs [hunk 1]
```diff
@@ -42,7 +42,7 @@ public async Task WriteAsync_Works()
 
         //Assert
         stream.Position = 0;
-        var problemDetails = await JsonSerializer.DeserializeAsync<ProblemDetails>(stream);
+        var problemDetails = await JsonSerializer.DeserializeAsync<ProblemDetails>(stream, new JsonOptions().JsonSerializerOptions);
```
Comment (captainsafia) [inline src/Mvc/Mvc.Core/test/Infrastructure/DefaultApiProblemDetailsWriterTest.cs:45]:
```suggestion
        var problemDetails = await JsonSerializer.DeserializeAsync<ProblemDetails>(stream, new JsonSerializerOptions(JsonSerializerDefaults.Web));
```

File: src/Http/Http.Results/test/ProblemResultTests.cs [hunk 2]
```diff
@@ -11,6 +12,8 @@ namespace Microsoft.AspNetCore.Http.HttpResults;
 
 public class ProblemResultTests
 {
+    private static readonly JsonSerializerOptions SerializerOptions = new JsonOptions().SerializerOptions;
```
Comment (captainsafia) [inline src/Http/Http.Results/test/ProblemResultTests.cs:15]:
```suggestion
    private static readonly JsonSerializerOptions SerializerOptions = new JsonSerializerOptions(JsonSerializerDefaults.Web);
```

File: src/Http/Http.Results/test/JsonResultTests.cs [hunk 3]
```diff
@@ -12,6 +14,8 @@ namespace Microsoft.AspNetCore.Http.HttpResults;
 
 public class JsonResultTests
 {
+    private static readonly JsonSerializerOptions SerializerOptions = new JsonOptions().SerializerOptions;
```
Comment (captainsafia) [inline src/Http/Http.Results/test/JsonResultTests.cs:17]:
```suggestion
    private static readonly JsonSerializerOptions SerializerOptions = new JsonSerializerOptions(JsonSerializerDefaults.Web);
```

**Review by captainsafia — APPROVED** [review]
I notice that we use different serializer options across different test cases. Should we standardize on the web defaults for everything?

**Review by halter73 — APPROVED** [review]

**Pull request merged**